services/core/PlatformDriverAgent/HomeAssistantDriver.md

```diff
@@ -0,0 +1,239 @@
+
+```
+The first thing you will need is the IP address and your long lived access token from your Home Assistant instance. 
+In the Home Assistant ui you can click on your profile picture and scroll all the way down to find your long lived access token. 
+![image](https://github.com/riley206/Rileys_volttron/assets/89715390/deb8d632-b0db-45de-ac7d-88c27e5fb2a1)
+
+
+Next, you can either clone this repository as is, or manually grab the three files necessary to add to your VOLTTRON instance.
+
+The three files you will need are HomeAssistant_Driver/home_assistant.config, HomeAssistant_Driver/home_assistant_registers.json, and services/core/PlatformDriverAgent/platform_driver/interfaces/home_assistant.py
+
+Once you have these files in the same place (or you just cloned this) you can add what you need to the config file and the registers file. The config file stores your IP address, your port, and your access token from Home Assistant. As of now please keep the VOLTTRON topic the same, this allows for the detection of changes on the message bus. Will be fixed later. 
+
+```json
+{
+    "driver_config": {
+        "ip_address": "Your Home Assistant IP",
+        "access_token": "Your Home Assistant Access Token",
+        "volttron_topic": "your volttron topic",
+        "port": "Your Port"
+    },
+    "driver_type": "home_assistant",
+    "registry_config":"config://home_assistant_registers.json",
+    "interval": 30,
+    "timezone": "UTC"
+}
+```
+Your register file will can contain multiple entities but is best used with one. Using one keeps everything more organized.
+
+```json
+    {
+        "Point Name": "sensor.average_humidity_1621",
+        "Volttron Point Name": "sensor.average_humidity_1621",
+        "Units": "%",
+        "Units Details": "%",
+        "Writable": true,
+        "Starting Value": 20,
+        "Type": "float",
+        "Notes": "Average humidity of 1621",
+    },
+```
+
+Other devices such as thermostats will have multiple attributes, simply add the attributes in the attributes field.
+
+```json
+    {
+        "Point Name": "climate.thermostat",
+        "Volttron Point Name": "climate.thermostat",
+        "Units": "C",
+        "Units Details": "C",
+        "Writable": true,
+        "Starting Value": true,
+        "Type": "boolean",
+        "Notes": "lights bedroom",
+        "Attributes": {
+            "current_temperature": "current_temperature",
+            "current_humidity": "current_humidity"
+        }
+    },
+```
+
+
+Attributes can be found in developer tools or by opening the device in the GUI of Home Assistant. 
+
+![image](https://github.com/riley206/Rileys_volttron/assets/89715390/a367e61e-8b73-4f35-a179-dfda235ddcbe)
+
+
+We are now ready to get the driver running. Right now you can install the listener agent and a historian if you are looking to store data and view in VOLTTRON. Install the platform driver and make sure the home_assistant.py is in the interfaces folder. 
```

Comment:
Use a listener agent to validate that the driver is publishing properly.  Note the home_assistant will be in the correct directory because they have this code.